source/adios2/common/ADIOSTypes.cpp

```diff
@@ -367,4 +367,42 @@ void MinMaxStruct::Dump(DataType Type)
         break;
     }
 }
+
+int TypeElementSize(DataType adiosvartype)
```

Comment:
Is there a reason why we are hardcoding this and not using `sizeof(uint8_t)` etc?

Reply:
No great reason, except it was shorter to type.  ADIOS wouldn't do well if the size implications of any of the types (including float and double) ever changed.   (Which kind of generically excludes machines like the old KSR-1 where float was 8 bytes and double was 16.)  Looking at the level of type flexibility that would let us move data between machines with different underlying data representations is an interesting topic though...

Reply:
I guess this is not an issue for everywhere we run ADIOS2 now, but it would be good to think about it before we are forced to do it :) (not part of this PR)